javascript/ql/experimental/adaptivethreatmodeling/src/StoredXssATM.ql

```diff
@@ -0,0 +1,27 @@
+/**
+ * For internal use only.
+ *
+ * @name Stored cross-site scripting (experimental)
+ * @description Using uncontrolled stored values in HTML allows for a stored cross-site scripting vulnerability.
+ * @kind path-problem
+ * @scored
+ * @problem.severity error
+ * @security-severity 6.1
+ * @id js/ml-powered/stored-xss
+ * @tags experimental security external/cwe/cwe-079 external/cwe/cwe-116
+ */
+
+import experimental.adaptivethreatmodeling.StoredXssATM
+import ATM::ResultsInfo
+import DataFlow::PathGraph
+
+from
+  DataFlow::Configuration cfg, DataFlow::PathNode source, DataFlow::PathNode sink, float score,
+  string scoreString
+where
+  cfg.hasFlowPath(source, sink) and
+  not isFlowLikelyInBaseQuery(source.getNode(), sink.getNode()) and
+  score = getScoreForFlow(source.getNode(), sink.getNode()) and
+  scoreString = getScoreStringForFlow(source.getNode(), sink.getNode())
+select sink.getNode(), source, sink,
+  "(Experimental) This may be a stored cross-site scripting vulnerability due to $@. Identified using machine learning."
```

Comment:
This is missing the placeholder (the thing to fill in `$@` in the original message with).  Try adding `source.getNode(), "stored value"` to the end of the select clause (as with the unboosted query).

Reply:
Like this?
![Screenshot 2022-04-27 at 15 45 48](https://user-images.githubusercontent.com/14887813/165545612-865e9622-e4db-4d24-ba33-54de76356746.png)

Reply:
Looks good!